### PR TITLE
`epoll` with fs fixes

### PR DESCRIPTION
**Pull request by john-sharratt**

### CODE REVIEW

**Comment by ptitSeb** [comment]
@john-sharratt so this one replace completly https://github.com/wasmerio/wasmer/pull/3830 ?

**Comment by john-sharratt** [comment]
Need to move over to this one
https://github.com/wasmerio/wasmer/pull/4060

**Comment by john-sharratt** [comment]
Now that the problem is located, will move forward with this branch again

**Comment by Michael-F-Bryan** [comment]
@ptitSeb and @john-sharratt I've just pushed [409b1d0](https://github.com/wasmerio/wasmer/pull/4050/commits/409b1d0bdf22d4ce388ca2dec498f2efede430e8) to make sure we initialise the `PluggableRuntime` with a `PackageLoader`. We don't use `BuiltinPackageLoader` by default any more (see https://github.com/wasmerio/wasmer/pull/4055), so it needs to be configured manually.

File: lib/virtual-fs/src/host_fs.rs [hunk 1]
```diff
@@ -428,6 +455,7 @@ impl AsyncRead for File {
         cx: &mut Context<'_>,
         buf: &mut tokio::io::ReadBuf<'_>,
     ) -> Poll<io::Result<()>> {
+        let _guard = Handle::try_current().map_err(|_| self.handle.enter());
```
[the remark this answers is not in the document]
Comment (john-sharratt) [reply lib/virtual-fs/src/host_fs.rs:458]: as discussed, this is the least evil or the evils

File: examples/wasi_manual_setup.rs [hunk 2]
```diff
@@ -33,6 +33,13 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // Let's compile the Wasm module.
     let module = Module::new(&store, wasm_bytes)?;
 
+    println!("Starting `tokio` runtime...");
+    let runtime = tokio::runtime::Builder::new_multi_thread()
```
[the remark this answers is not in the document]
Comment (john-sharratt) [reply examples/wasi_manual_setup.rs:37]: as discussed, all the inner runtimes are now gone

**Comment by john-sharratt** [comment]
wasi::python test is flaky as its dependent on external API calls (GraphQL)

**Comment by john-sharratt** [comment]
I fixed up the Wasmer Web build - next up the snapshot tests seem to still be failing

File: examples/wasi_manual_setup.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (syrusakbary) [reply examples/wasi_manual_setup.rs:37]: You probably want to pass this to the builder via `runtime(Arc::new(rt));` ?

[the remark this answers is not in the document]
Comment (john-sharratt) [reply examples/wasi_manual_setup.rs:37]: That would too many breaking changes in the API as the runtime would need to be plumbed through lots of structs until it reaches the virtual net objects which themselves are concrete classes anyway.... for another time.

**Review by ptitSeb — APPROVED** [review]

**Pull request merged**